### ISSUE TITLE
Vijay Midterm Comments

### ISSUE THREAD

**Issue by ostegm** [ga-students/DAT_SF_12#13]
Here are a few comments. Let any of the TA's know if you have questions. 

Exercise 1 - Data Exploration (10/15 points):
++++++++++++++++++++
-Good use of SNS pairplot to explore the linear relationships, however I didn't see a lot of discussion about which vairables might be explanatory. 
-It would have been good to see a few vairables pulled out a mapped with spam vs not spam using color coding. Something like a boxplot could work well:

```
#limit df to non-outliers to see the boxplot and visualize a boxplot
df1 = spam_data[spam_data['capital_run_length_longest'] < 500]
sns.boxplot(df1['capital_run_length_longest'], groupby = df1['is_spam'])
```

Exercise 2 - Apply logistic Regression: (15/15 points)
++++++++++++++++++++++++++++++++++++++++
-Great application of logistic regression. It looks like you pulled out one of the most explanatory variable (capital run length longest). 
-Excellent application of the thresholding idea! You did an awesome job at building out predict_data dataframe! 
-Also, good use of the Sklearn toolkit! it was cool to see how you did that despite not having built the model in sklearn. Good job! 

Exercise 3 - Train test split: 5/5 points
++++++++++++++++++++++++++++++++++++++++
-Perfect! No issues here! 
-I see that you understand the data set well and understand the concepts. Good work. 

Exercise 4 - Cross-validation: 5/5 points
++++++++++++++++++++++++++++++++++++++++
-Nice work transitioning to sklearn for this. It would have been difficult to build your own CV scoring method, but its possible! 
-30 folds is a lot, but it worked will in this case. In other cases where you have less data, using 30 folds could be too many, but no issues here. 

Exercise 5 - Compare to KNN:  12/15 points
++++++++++++++++++++++++++++++++++++++++
-You applied KNN correctly, but then found that the accuracy improves at n=5. It seems like it would make sense to use n=5 instead of n=3 to compare model performance, right? 
-If we are doing model selection and you find one model is better, that's great, but we need to make sure we optimize each model when comparing. 
-But otherwise good job! 

Exercise 6 - Evaluate Feature Importance: 15 points
++++++++++++++++++++++++++++++++++++++++++++++++
-Good use of p values to show feature significance. It would be great to show this visually too. Maybe a simple bar chat? 

Exercise 7 - Plot ROC Curve: 10 points
++++++++++++++++++++++++++++++++++++++++

Exercise 8 - Test various thresholds: 10 points
++++++++++++++++++++++++++++++++++++++++

Exercise 9 - Discussion of thresholds: 10 points
++++++++++++++++++++++++++++++++++++++++

Bonus - Apply PCA and visualize the data in two dimensions: 20 points
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
